api/src/main/java/org/openmrs/module/pihapps/PihAppsServiceImpl.java

```diff
@@ -68,4 +100,123 @@ public void updateVisitAndLoginLocations(List<Location> visitLocations, List<Loc
+	}
+
+	@Override
+	@Transactional(readOnly = true)
+	@Authorized(PrivilegeConstants.GET_ORDERS)
+	@SuppressWarnings({ "unchecked", "deprecation" })
+	public LabOrderSearchResult getLabOrders(LabOrderSearchCriteria searchCriteria) {
+		LabOrderSearchResult result = new LabOrderSearchResult();
+		result.setCriteria(searchCriteria);
+
+		Date now = new Date();
+
+		Criteria c = sessionFactory.getHibernateSessionFactory().getCurrentSession().createCriteria(Order.class);
+		c.add(eq("voided", false));
+		c.add(in("orderType", getLabOrderTypes(true))); // Limit to lab order types
+		c.add(in("action", Order.Action.NEW, Order.Action.REVISE, Order.Action.RENEW)); // Exclude DC orders
+
+		if (searchCriteria.getPatient() != null) {
+			c.add(eq("patient", searchCriteria.getPatient()));
+		}
+		if (searchCriteria.getLabTest() != null) {
+			c.add(eq("concept", searchCriteria.getLabTest()));
+		}
+		if (StringUtils.isNotBlank(searchCriteria.getAccessionNumber())) {
+			c.add(eq("accessionNumber", searchCriteria.getAccessionNumber()).ignoreCase());
+		}
+		if (searchCriteria.getActivatedOnOrBefore() != null) {
+			Date onOrBefore = OpenmrsUtil.getLastMomentOfDay(searchCriteria.getActivatedOnOrBefore());
+			c.add(le("dateActivated", onOrBefore));
+		}
+		if (searchCriteria.getActivatedOnOrAfter() != null) {
+			Date onOrAfter = OpenmrsUtil.firstSecondOfDay(searchCriteria.getActivatedOnOrAfter());
+			c.add(ge("dateActivated", onOrAfter));
+		}
+		if (searchCriteria.getOrderStatus() != null && !searchCriteria.getOrderStatus().isEmpty()) {
+			Criterion[] orderStatusCriteria = new Criterion[searchCriteria.getOrderStatus().size()];
+			for (int i = 0; i < searchCriteria.getOrderStatus().size(); i++) {
+				OrderStatus orderStatus = searchCriteria.getOrderStatus().get(i);
+				if (orderStatus == OrderStatus.ACTIVE) {
+					Criterion orderIsNotExpired = or(isNull("autoExpireDate"), gt("autoExpireDate", now));
+					Criterion orderIsNotStopped = or(isNull("dateStopped")); // This should never be in the future
+					orderStatusCriteria[i] = and(orderIsNotExpired, orderIsNotStopped);
+				}
+				else if (orderStatus == OrderStatus.EXPIRED) {
+					orderStatusCriteria[i] = le("autoExpireDate", now);
+				}
+				else if (orderStatus == OrderStatus.STOPPED) {
+					orderStatusCriteria[i] = isNotNull("dateStopped");
+				}
+			}
+			c.add(or(orderStatusCriteria));
+		}
+		List<Criterion> fulfillerStatusCriteria = new ArrayList<>();
+		if (searchCriteria.getFulfillerStatuses() != null && !searchCriteria.getFulfillerStatuses().isEmpty()) {
+			for (Order.FulfillerStatus fulfillerStatus : searchCriteria.getFulfillerStatuses()) {
+				fulfillerStatusCriteria.add(eq("fulfillerStatus", fulfillerStatus));
+				// TODO: For all intents and purposes, we equate a null fulfillerStatus and a RECEIVED fulfillerStatus to mean ordered but not in progress
+				if (fulfillerStatus == Order.FulfillerStatus.RECEIVED) {
+					fulfillerStatusCriteria.add(isNull("fulfillerStatus"));
+				}
```

Comment:
I'm probably missing something but why do we need to handle this at the service layer?  I would think we should just search for exactly what is asked for here?  Do we currently use RECEIVED at all in our app?  I would think if the filter set 'Ordered', we'd just set the fulfiller status criteria to null/empty?